pkg/gadgets/internal/socketenricher/bpf/sockets-map.bpf.c

```diff
@@ -130,6 +130,34 @@ int ig_sockets_it(struct bpf_iter__task_file *ctx)
 	return 0;
 }
 
+// This iterator is called from a Go Ticker to remove expired sockets
+SEC("iter/bpf_map_elem")
+int ig_sk_cleanup(struct bpf_iter__bpf_map_elem *ctx)
+{
+	struct seq_file *seq = ctx->meta->seq;
+	__u32 seq_num = ctx->meta->seq_num;
+	struct bpf_map *map = ctx->map;
+	struct sockets_key *socket_key = ctx->key;
+	struct sockets_key tmp_key;
+	struct sockets_value *socket_value = ctx->value;
+
+	if (!socket_key || !socket_value)
+		return 0;
+
+	__u64 now = bpf_ktime_get_boot_ns();
+	__u64 deletion_timestamp = socket_value->deletion_timestamp;
+	__u64 socket_expiration_ns = 1000ULL*1000ULL*1000ULL*5ULL; // 5 seconds
+
+	if (deletion_timestamp != 0 && deletion_timestamp + socket_expiration_ns < now) {
+        // The socket is expired, remove it from the map.
+        __builtin_memcpy(&tmp_key, socket_key, sizeof(struct sockets_key));
+        bpf_map_delete_elem(&sockets, &tmp_key);
```

Comment:
Does the verifier complains if you pass `socket_key` directly to `bpf_map_delete_elem`?

Reply:
Yes, the verifier failed.
The same technique is used in this sample code:
https://github.com/torvalds/linux/blob/v6.3/tools/testing/selftests/bpf/progs/bpf_iter_bpf_hash_map.c#L70